### DISCUSSION TITLE
Inline editor for M2O fields

### DISCUSSION THREAD

**Discussion by sebescudie** [directus/directus#16029] [Feature Requests]
Hey all, 

This conversation started on the Discord server. The idea would be to have some sort of "inline" editor for M2O fields. Here's my use case :

- I have a bunch of singleton tables that should all share some common fields, and should also all introduce their own custom fields. The structure of those custom fields could vary drastically between singletons.
- So far, I tackled this by creating a "base" collection that I add as an M2O field in all my singletons. This base collection contains all common fields that all singletons should have. After that, I add all my custom fields in each singleton.

I was looking for a way to let the user edit that M2O field directly on the singleton page, without having to click on the `+` button and create/edit the field in the right panel. I would like those "base" fields to show up on the singleton page as if they were part of it.

Same as we have a _Layout_ dropdown when creating an O2M/M2M (allowing to pick the list or table layout), I would suggest having the same dropdown for M2O with an _Inline_ option. Picking that option would result in the behavior I described above : the fields of the M2O are directly editable on the "parent's" page.

What do you think?

Thanks in advance!

**Comment by rijkvanzanten** [comment]
Would that still be a "m2o" where you'd be able to select a pre-existing item, and then edit that inline, or would it be more like a "one to one" where the fields are presented inline, but you're not able to reuse existing items?

**Reply by sebescudie** [reply]
Hey there, thanks for your answer. What I have in mind here is a "one to one" scenario.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
